scipy/ndimage/tests/test_interpolation.py

```diff
@@ -167,12 +203,16 @@ def mapping(x):
         assert_array_almost_equal(out, [0, 4, 1, 3])
 
     @pytest.mark.parametrize('order', range(0, 6))
-    @pytest.mark.parametrize('dtype', [np.float64, np.complex128])
-    def test_geometric_transform05(self, order, dtype):
-        data = np.array([[1, 1, 1, 1],
+    @pytest.mark.parametrize('dtype', ["float64", "complex128"])
+    def test_geometric_transform05(self, order, dtype, xp):
+        if is_cupy(xp):
+            pytest.xfail("CuPy does not have geometric_transform")
+
+        dtype = getattr(xp, dtype)
+        data = xp.asarray([[1, 1, 1, 1],
                          [1, 1, 1, 1],
                          [1, 1, 1, 1]], dtype=dtype)
```

Comment:
```suggestion
        data = xp.asarray([[1, 1, 1, 1],
                           [1, 1, 1, 1],
                           [1, 1, 1, 1]], dtype=dtype)
```

and below in various places